plugins/modules/jenkins_credential.py

```diff
@@ -0,0 +1,863 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: jenkins_credential
+short_description: Manage Jenkins credentials and domains via API
+version_added: 11.1.0
+description:
+  - This module allows managing Jenkins credentials and domain scopes via the Jenkins HTTP API.
+  - Create, update, and delete different credential types such as C(username/password), C(secret text), C(SSH key), C(certificates), C(GitHub App), and domains.
+  - For scoped domains (O(type=scope)), it supports restrictions based on V(hostname), V(hostname:port), V(path), and V(scheme).
+requirements:
+  - urllib3 >= 1.26.0
+author:
+  - Youssef Ali (@YoussefKhalidAli)
+extends_documentation_fragment:
+  - community.general.attributes
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+options:
+  id:
+    description:
+      - The ID of the Jenkins credential or domain.
+    type: str
+  type:
+    description:
+      - Type of the credential or action.
+    choices:
+      - user_and_pass
+      - file
+      - text
+      - github_app
+      - ssh_key
+      - certificate
+      - scope
+      - token
+    type: str
+  state:
+    description:
+      - The state of the credential.
+    choices:
+      - present
+      - absent
+    default: present
+    type: str
+  scope:
+    description:
+      - Jenkins credential domain scope.
+      - Deleting a domain scope deletes all credentials within it.
+    type: str
+    default: '_'
+  force:
+    description:
+      - Force update if the credential already exists, used with O(state=present).
+      - If set to V(true), it deletes the existing credential before creating a new one.
+      - Always returns RV(ignore:changed=true).
+    type: bool
+    default: false
+  url:
+    description:
+      - Jenkins server URL.
+    type: str
+    default: http://localhost:8080
+  jenkins_user:
+    description:
+      - Jenkins user for authentication.
+    required: true
+    type: str
+  jenkins_password:
+    description:
+      - Jenkins password for token creation. Required if O(type=token).
+    type: str
+  token:
+    description:
+      - Jenkins API token. Required unless O(type=token).
+    type: str
+  description:
+    description:
+      - Description of the credential or domain.
+    default: ''
+    type: str
+  location:
+    description:
+      - Location of the credential. Either V(system) or V(folder).
+      - If O(location=folder) then O(url) must be set to V(<jenkins-server>/job/<folder_name>).
+    choices:
+      - system
+      - folder
+    default: 'system'
+    type: str
+  name:
+    description:
+      - Name of the token to generate. Required if O(type=token).
+      - When generating a new token, don't pass O(id). It is generated automatically.
```

Comment:
Please do not use contractions in the docs.
```suggestion
      - When generating a new token, do not pass O(id). It is generated automatically.
```